splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java

```diff
@@ -35,35 +34,40 @@
     static final CurrentNetwork NO_NETWORK = new CurrentNetwork(NetworkState.NO_NETWORK_AVAILABLE, null);
     static final CurrentNetwork UNKNOWN_NETWORK = new CurrentNetwork(NetworkState.TRANSPORT_UNKNOWN, null);
 
-    private final ConnectionMonitor connectionMonitor;
+    private final Supplier<NetworkRequest> createNetworkMonitoringRequest;
     private final NetworkDetector networkDetector;
+    private final ConnectivityManager connectivityManager;
 
-    private final AtomicReference<CurrentNetwork> currentNetwork = new AtomicReference<>();
+    private volatile CurrentNetwork currentNetwork;
```

Comment:
We don't use CAS anywhere, so I replaced an `AtomicReference` with a `volatile` field